### PR TITLE
refactor(popconfirm): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  refactor with hook         |
| 🇨🇳 Chinese |  用 react hook 重构         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/popconfirm/index.tsx [hunk 1]
```diff
@@ -31,136 +31,106 @@ export interface PopconfirmLocale {
...
+      placement={placement}
+      onVisibleChange={onVisibleChange}
+      visible={visible}
+      overlay={overlay}
+      ref={ref as any}
+    />
+  );
+});
+
+Popconfirm.defaultProps = {
```
[the remark this answers is not in the document]
Comment (hengkx) [reply components/popconfirm/index.tsx:127]:
```jsx
React.forwardRef<unknown, PopconfirmProps>((({ placement='top'}), ref)=>{})
```
这样？

[the remark this answers is not in the document]
Comment (orzyyyy) [reply components/popconfirm/index.tsx:127]: 我是指 defaultProps 的定义可以删了，没必要再弄一套属性定义。

[the remark this answers is not in the document]
Comment (afc163) [reply components/popconfirm/index.tsx:127]: defaultProps 不是 propTypes

[the remark this answers is not in the document]
Comment (hengkx) [reply components/popconfirm/index.tsx:127]: > 我是指 defaultProps 的定义可以删了，没必要再弄一套属性定义。

我知道你的意思是删除`defaultProps`, 那得有地方给默认赋值把？

[the remark this answers is not in the document]
Comment (orzyyyy) [reply components/popconfirm/index.tsx:127]: 是我看错了

**Comment by hengkx** [comment]
@afc163 @orzyyyy @AshoneA  review一下 看下没有问题合一下，然后合并进feature分支，我把这个问题处理一下 #20249

**Pull request merged**